### PR TITLE
More cockpit.defer() cleanup

### PR DESCRIPTION
**Pull request by martinpitt**
Only three usages left!
 - The notify/progress stuff in cockpit-components-install-dialog.jsx ; after that, we can drop it from pkg/playground/react-patterns.js
 - pkg/base1/test-dbus.js: D-Bus method wants to return a tuple of values, which doesn't work as a list; needs some thinking and API work
 - pkg/lib/journal.js: Slaps some custom .stream() API onto the promise, this needs a more careful look.

### CODE REVIEW

File: pkg/users/authorized-keys.js [hunk 1]
```diff
@@ -79,12 +77,12 @@ function AuthorizedKeys (user_name, home_dir) {
                                         .join(" ") || null;
                         }
                     }
+                    return keys;
                 })
-                .always(function() {
-                    df.resolve(keys);
+                .catch(ex => {
+                    cockpit.warn("Failed to list public keys:", ex.toString());
+                    return [];
```
Comment (cockpituous) [inline pkg/users/authorized-keys.js:84]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18488-20230310-163728-a6731aad-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys.js [hunk 2]
```diff
@@ -97,34 +95,23 @@
...
                                 .script(adder, [user_name, home_dir], { superuser: "try", err: "message" })
                                 .input(obj.raw + "\n")
-                                .done(function() {
-                                    df.resolve();
-                                })
-                                .fail(function(ex) {
-                                    df.reject(_("Error saving authorized keys: ") + ex);
-                                });
+                                // eslint-disable-next-line prefer-promise-reject-errors
+                                .catch(ex => Promise.reject(_("Error saving authorized keys: ") + ex));
```
Comment (cockpituous) [inline pkg/users/authorized-keys.js:110]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18488-20230310-163728-a6731aad-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Argh, this needs some retries. 3x affected retries on notoriously unstable networking tests..

File: pkg/users/authorized-keys.js (hunk 1 above)
Comment (cockpituous) [inline pkg/users/authorized-keys.js:84]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18488-20230310-180230-a6731aad-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys.js [hunk 3]
```diff
@@ -97,34 +95,23 @@ function AuthorizedKeys (user_name, home_dir) {
...
```
Comment (cockpituous) [inline pkg/users/authorized-keys.js:110]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18488-20230310-180230-a6731aad-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
:+1:

**Pull request merged**